_posts/2017-02-21-Rails-5-1-beta1.markdown

```diff
@@ -0,0 +1,108 @@
+---
+layout: post
+title: 'Rails 5.1.0.beta1: Loving JavaScript, System Tests, Encrypted Secrets, and more'
+category: releases
+author: dhh
+published: true
+date: 2017-02-21 20:00:00 +01:00
+---
+Rails 5.0 was released just some eight months ago, and now, [some 3500 commits later](https://github.com/rails/rails/compare/v5.0.0...master/), we're already close to the next big release. And what release this version 5.1 is lining up to be! We've made great strides on long-term promises and key ergonomics while also spring cleaning a bunch of deprecated code. 
+
+Let me walk you through the highlight reel:
+
+
+## Loving JavaScript
+
+We've had a stormy, perhaps even contentious, relationship with JavaScript over the years. But that time is past. JavaScript has improved immensely over the past few years, particularly with the advent of ES6, and with package and compilation tools like Yarn and Webpack. Rails is embracing both of these solutions with open arms and letting whatever past water flow under the bridge.
+
+JavaScript and Ruby share a deep philosophical bond over language design, if not ecosystem management. Let's focus on the aspects we have in common and help Rails programmers extract the best from JavaScript with the help of some key guiding conventions.
+
+The improvements in Rails 5.1 focus on three major parts:
+
+1. **Manage JavaScript dependencies from NPM via Yarn.** Think of [Yarn](https://yarnpkg.com/en/) like Bundler for JavaScript (it even has Yehuda Katz involved!). This makes it easy to depend on libraries like React or anything else from NPM. Everything you depend on via Yarn is then made available to be required in the asset pipeline, just like vendored dependencies would have been. Just use the binstub bin/yarn to add dependencies.
+
+2. **Optionally compile JavaScript with Webpack.** While there are a million different module bundlers/compilers for JavaScript, [Webpack](https://webpack.github.io/) is quickly emerging as the preeminent choice. We've made it easy to use Webpack with Rails through the new [Webpacker](https://github.com/rails/webpacker) gem that you can configure automatically on new projects with --webpack. This is fully compatible with the asset pipeline, which you can continue to use for images, fonts, sounds, whatever. You can even have some JavaScript on the asset pipeline and some done via Webpack. It's all managed via Yarn that's on by default.
+
+3. **Drop jQuery as a default dependency.** We used to require jQuery in order to provide features like data-remote, data-confirm, and the other parts of Rails UJS. This dependency is no longer necessary as we've rewritten rails-ujs to use vanilla JavaScript. You're of course still free to use jQuery, but you no longer have to.
+
+
+Thanks to [Liceth Ovalles](https://github.com/Liceth) for her work on Yarn integration and [Guillermo Iguaran](https://github.com/guilleiguaran) for his work on rails-ujs, Yarn, and Webpack!
```

Comment:
Sure, we should mention to Dangyi Liu (https://github.com/liudangyi) here also!!